packages/sources/tiingo/src/index.ts

```diff
@@ -1,5 +1,7 @@
 import { expose } from '@chainlink/ea-bootstrap'
 import { makeExecute, makeWSHandler } from './adapter'
 import { NAME, makeConfig } from './config'
+import { ResponseSchema } from './endpoint/crypto/top'
```

Comment:
What if we want to export a different ResponseSchema from a different endpoint?

Reply:
Can we do something like this so we can support ResponseSchema from multiple different endpoints?

```
import * as CryptoTop from './endpoint/crypto/top'
```

Reply:
do we then want to export everything from `CryptoTop`?

Reply:
Even better might be to come up with a data structure for exporting types. We want this across all adapters that will be exported like:

```
export { NAME, makeExecute, makeConfig, types }
```

Something like:
```
{
endpoints: {
  [name: string]: {
 ResponseSchema
}
},
config
}
```

Reply:
@HenryNguyen5 curious on your thoughts on how to wrangle types

Reply:
I would structure the directory tree in the following way:
``` ts
// endpoints.ts
export * as foo from './foo'
export * as bar from './'bar'
```

```ts
// endpoints/foo/index.ts
import {ResponseSchema} from '../types'
export * as baz from './baz'
export const x: ResponseSchema = {...}
export const y: ResponseSchema = {...}
```

```ts
// endpoints/foo/baz/index.ts
import {ResponseSchema} from '../../types'
export const z: ResponseSchema = {...}
```

```ts
// endpoints/bar/index.ts
import {ResponseSchema} from '../types'
export const x: ResponseSchema = {...}
export const y: ResponseSchema = {...}
```

Then we can import them as the following:
``` ts
// index.ts
import * as endpoints from './endpoints'
endpoints.foo.x()
endpoints.foo.baz.z()
endpoints.bar.y()
```

Reply:
@HenryNguyen5 are you suggesting that we define the `ResponseSchema` interface in a types file and then export it from there?  The `ResponseSchema` interface is different for each endpoint though.  Should each endpoint have it's own types file in it's directory?

Reply:
Ah, didnt realize that.
Yeah, each endpoint can define its own types, such as `PriceResponse`, `CryptoTopResponse`, rather than a generic `ResponseSchema`

Reply:
Alright sounds good.  Then in the adapter's index file we just import everything from the `endpoints` and just export that as `types`?

Reply:
You can re-export the types from within their relevant endpoint files, then they'd be bound to that module scope.
```ts
// endpoints/foo
export * from './types'
import * as responses from './types'
export const crypto: responses.CryptoResponse = () => {}
```
```ts
// endpoints/foo/types.ts
export interface CryptoResponse {}
```
```ts
// index
import * as endpoints from './endpoints'
type CryptoResponse = endpoints.foo.CryptoResponse
const cryptoResponse = await endpoints.foo.crypto(af
```